pkg/restore/restore.go

```diff
@@ -363,6 +419,41 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 	return warnings, errs
 }
 
+// getNamespace returns a namespace API object that we should attempt to
+// create before restoring anything into it. It will come from the backup
+// tarball if it exists, else will be a new one. If from the tarball, it
+// will retain its labels, annotations, and spec.
+func getNamespace(path, remappedName string) *v1.Namespace {
+	var nsBytes []byte
+	var err error
+
+	if nsBytes, err = ioutil.ReadFile(path); err != nil {
+		return &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: remappedName,
+			},
+		}
+	}
+
+	var backupNS v1.Namespace
+	if err := json.Unmarshal(nsBytes, &backupNS); err != nil {
+		return &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: remappedName,
+			},
+		}
+	}
+
+	return &v1.Namespace{
```

Comment:
Would it make sense to start with `backupNS` and mutate fields as needed? That would ensure we get everything from the backed up NS, instead of "opting in" to just labels & annotations.

Reply:
hmm. not sure. with other objects we're restoring, in the `resetMetadataAndStatus` func we are taking the opt-in approach. If we take the alternate approach there's a whole bunch of stuff that we have to explicitly clear out of ObjectMeta (`selfLink`, `uid`, `creationTimestamp`, etc.)